filters/include/pcl/filters/voxel_grid_covariance.h

```diff
@@ -425,7 +454,13 @@ namespace pcl
         k_leaves.reserve (k);
         for (const int &k_index : k_indices)
         {
-          k_leaves.push_back (&leaves_[voxel_centroids_leaf_indices_[k_index]]);
+          auto voxel = leaves_.find(voxel_centroids_leaf_indices_[k_index]);
+          if (voxel == leaves_.end()) {
+            PCL_WARN("Accessing to non-existent voxel");
```

Comment:
I would remove this warning. The policy for showing warnings is limited to cases where you break a user's expectation somehow and is really important that he is aware of it.

That being said and before doing anything here, how realistic is it to trigger this condition. Is this supposed to ever happen. I see you're using indices supplied by a kdtree search.

Reply:
I removed the warning.

I think this condition never happens unless ```kdtree_``` is modified from outside of the class, because ```kdtree_``` is built from ```leaves_```, and thus every point in ```kdtree_``` should have a corresponding voxel in ```leaves_```. 

https://github.com/PointCloudLibrary/pcl/blob/6aaade34381fbb608208172e623f71719565fd33/filters/include/pcl/filters/voxel_grid_covariance.h#L274
https://github.com/PointCloudLibrary/pcl/blob/6aaade34381fbb608208172e623f71719565fd33/filters/include/pcl/filters/voxel_grid_covariance.h#L269
https://github.com/PointCloudLibrary/pcl/blob/6aaade34381fbb608208172e623f71719565fd33/filters/include/pcl/filters/impl/voxel_grid_covariance.hpp#L306